src/tree/node_children.rs

```diff
@@ -560,33 +559,33 @@ mod inner {
         /// Access to the nodes array.
         #[inline(always)]
         pub fn nodes<'a>(&'a self) -> &'a [Arc<Node>] {
-            unsafe { mem::transmute(&self.nodes[..(self.len())]) }
+            unsafe { &*(&self.nodes[..(self.len())] as *const _ as *const _) }
```

Comment:
So, I realize that Clippy harps on this one a lot.  But it also makes it really hard to see what's going on, IMO, and isn't meaningfully safer.  Unless I'm missing something serious about guarding against safety issues, I'd rather leave this one out.